### PR TITLE
xe-guest-utilities: init at 6.2.0

### PR DESCRIPTION
**Pull request by benwbooth**
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

cc @benwbooth
# Notes:

I'm using NixOS at work, where we use Citrix Xen as our VM Host. The xe-guest-utilities allows me to query the Xen host for IP addresses of VMs, a crucial feature for me, which is why I went through the trouble of packaging this.

I based this off of the [xe-guest-utilities](https://www.archlinux.org/packages/community/x86_64/xe-guest-utilities/) ArchLinux package. Citrix seems to require a login to download the files, so ArchLinux keeps a copy on their servers. This expression downloads the source from the Arch Linux servers, I'm not sure if that's a problem or if there's a better place to put the source files.

I did not add the services to the `nixos/modules/virtualisation/xen-domU.nix` and did not add the module to the `module-list.nix`. Since this is a new module I think it should be completely opt-in, at least until it gets more testing.

I've tested it so far on my NixOS VM, and I am able to query for IP addresses now. 

I had trouble running the `nox-review wip` command. I'm not sure if this is because I already made the commit before I ran the command. It says this:

```
==> We're in a git repo, trying to fetch it
error: The following untracked working tree files would be overwritten by checkout:
```

and then proceeds to list all the files in my local nixpkgs tree. This is my first package submission so I'm still learning the Nix way of doing things.

Please let me know if there's any corrections to be made or problems. Thanks!

### CODE REVIEW

File: nixos/modules/virtualisation/xe-guest-utilities.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.xe-guest-utilities;
+in {
+  options = {
+    services.xe-guest-utilities = {
```
Comment (jagajaga) [inline nixos/modules/virtualisation/xe-guest-utilities.nix]: Please use `mkEnableOption`.

File: pkgs/tools/virtualization/xe-guest-utilities/default.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  meta = {
+    description = "Citrix XenServer Tools";
+    homepage = http://citrix.com/English/ps2/products/product.asp?contentID=683148&ntref=hp_nav_US;
+    maintainers = with stdenv.lib.maintainers; [ benwbooth ];
+    platforms = stdenv.lib.platforms.linux;
+    license = [ stdenv.lib.licenses.gpl2 stdenv.lib.licenses.lgpl21 ];
+  };
+  src = fetchurl {
+    url = "https://sources.archlinux.org/other/community/xe-guest-utilities/xe-guest-utilities_${version}-1120.tar.gz";
+    md5 = "98644da204b18b695ede0ba45f4df22d";
```
Comment (zimbatm) [inline pkgs/tools/virtualization/xe-guest-utilities/default.nix]: can you switch this to use sha256 ? we are trying to get rid of md5

**Comment by fpletz** [comment]
> I did not add the services to the nixos/modules/virtualisation/xen-domU.nix and did not add the module to the module-list.nix

I think it's ok to add this module to `module-list.nix` because otherwise this service module can't be used easily. It isn't enabled by default so this shouldn't really matter.

Thanks! The code looks ok to me. If you add this module to `module-list.nix` and squash your commits, I think we can merge this and maybe later enabled this by default for Xen domUs. :+1:

**Comment by benwbooth** [comment]
Thanks for all the feedback! Please let me know if there's anything else I need to fix.

**Pull request merged**